Add null pointer check for evalues

JakeStevens opened the pull request:
Summary: The to* functions could de-reference a NULL pointer. Adding a check.

Differential Revision: D83742246

JacobSzwejbka commented on runtime/core/evalue.h:
```diff
@@ -335,6 +344,9 @@ struct EValue {
   executorch::aten::ArrayRef<double> toDoubleList() const {
     ET_CHECK_MSG(isDoubleList(), "EValue is not a Double List.");
+    ET_CHECK_MSG(
+        payload.copyable_union.as_double_list_ptr != nullptr,
```
We should check this in the ctors no?

JakeStevens commented:
@pytorchbot label "release notes: none"

JakeStevens committed:
Add null pointer check for evalues (#14749)

Summary:

The to* functions could de-reference a NULL pointer. Adding a check.

Differential Revision: D83742246
